core/sql/optimizer/ControlDB.cpp

```diff
@@ -108,7 +108,12 @@ ControlDB::~ControlDB()
 
 void ControlDB::setRequiredShape(ControlQueryShape *shape)
 {
-  delete requiredShape_;
+  if (requiredShape_)
+    {
+      delete requiredShape_->getShape();
+      delete requiredShape_;
```

Comment:
Though delete would also work fine because of the macro, I would prefer NADELETE here with the correct heap.